### ISSUE TITLE
tilt up not working for local development

### ISSUE THREAD

**Issue by CorsettiS** [astronomer/astronomer-cosmos#342]
When I tried to follow the guidelines of local development using tilt, I got the following error 

```
Loading Tiltfile at: ***/astronomer-cosmos/Tiltfile
Traceback (most recent call last):
  ***/astronomer-cosmos/Tiltfile:1:15: in <toplevel>
Error in docker_compose: command ["docker" "compose" "--project-name" "dev" "--project-directory" "***/astronomer-cosmos/dev" "-f" "***/astronomer-cosmos/dev/docker-compose.yaml" "config"] failed.
error: exit status 14
stdout: ""
stderr: 'open ***/astronomer-cosmos/dev/.env: no such file or directory
```

**Comment by jlaneve** [comment]
Can you try creating an empty .env file and see if that fixes the issue?

**Comment by CorsettiS** [comment]
@jlaneve it fixes it partially, however there are still few errors coming up
```

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
       → google-cloud-workflows 1.7.1 requires protobuf<4.0.0dev,>=3.19.0, but you have protobuf 4.23.3 which is incompatible.
       → google-cloud-vision 1.0.2 requires protobuf<4.0.0dev, but you have protobuf 4.23.3 which is incompatible.
       → google-cloud-videointelligence 1.16.3 requires protobuf<4.0.0dev, but you have protobuf 4.23.3 which is incompatible.
       → google-cloud-translate 1.7.2 requires protobuf<4.0.0dev, but you have protobuf 4.23.3 which is incompatible.
       → google-cloud-texttospeech 1.0.3 requires protobuf<4.0.0dev, but you have protobuf 4.23.3 which is incompatible.
       → google-cloud-tasks 2.10.1 requires protobuf<4.0.0dev,>=3.19.0, but you have protobuf 4.23.3 which is incompatible.
...
```

```
STEP 2/2 — Deploying
Container dev-airflow-init-1  Creating
Container dev-airflow-init-1  Created
no such service: redis

got unexpected error during build/deploy: command ["docker" "compose" "--project-name" "dev" "--project-directory" "***/astronomer-cosmos/dev" "-f" "***/astronomer-cosmos/dev/docker-compose.yaml" "up" "--no-deps" "-d" "airflow-init"] failed.
error: exit status 1

Build Failed: command ["docker" "compose" "--project-name" "dev" "--project-directory" "***/astronomer-cosmos/dev" "-f" "***/astronomer-cosmos/dev/docker-compose.yaml" "up" "--no-deps" "-d" "airflow-init"] failed.
error: exit status 1
```

**Comment by CorsettiS** [comment]
@jlaneve unfortunately there is still an error taking place, now on postgres side

<img width="1531" alt="Screenshot 2023-06-29 at 10 20 12" src="https://github.com/astronomer/astronomer-cosmos/assets/66136602/34652a3c-9ce1-4e32-8e86-52c0c8eb132e">

```
STEP 3/3 — Deploying
Container dev-airflow-init-1  Creating
Container dev-airflow-init-1  Created
no such service: postgres

got unexpected error during build/deploy: command ["docker" "compose" "--project-name" "dev" "--project-directory" "/***/astronomer-cosmos/dev" "-f" "/***/astronomer-cosmos/dev/docker-compose.yaml" "up" "--no-deps" "-d" "airflow-init"] failed.
error: exit status 1
```

The postgres container is up & running tho. when I access it, I obtain the following logs

```
PostgreSQL Database directory appears to contain a database; Skipping initialization


2023-06-29 08:19:43.922 UTC [1] LOG:  starting PostgreSQL 13.11 (Debian 13.11-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit

2023-06-29 08:19:43.922 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432

2023-06-29 08:19:43.922 UTC [1] LOG:  listening on IPv6 address "::", port 5432

2023-06-29 08:19:43.925 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"

2023-06-29 08:19:43.932 UTC [27] LOG:  database system was interrupted; last known up at 2023-06-29 08:16:01 UTC

2023-06-29 08:19:44.114 UTC [27] LOG:  database system was not properly shut down; automatic recovery in progress

2023-06-29 08:19:44.116 UTC [27] LOG:  redo starts at 0/15CFF48

2023-06-29 08:19:44.116 UTC [27] LOG:  invalid record length at 0/15CFF80: wanted 24, got 0

2023-06-29 08:19:44.116 UTC [27] LOG:  redo done at 0/15CFF48

2023-06-29 08:19:44.131 UTC [1] LOG:  database system is ready to accept connections
```

I then tried to run a simple **psql** command inside the container, but got the following error

```
# psql
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: FATAL:  role "root" does not exist
```

I am not a docker expert so it is hard for me to properly give a better feedback, sorry for that.

**Comment by tatiana** [comment]
@CorsettiS, which operating system are you using?

Are you able to connect to `psql` using the credentials declared in the Docker compose file?

**Comment by CorsettiS** [comment]
@tatiana I am, it connects fine. I am currently using a Mac. Instead of using **tilt**, I just manually did a docker compose up and it worked fine. I do not know how this tilt command works, but from what I saw it is breaking the build process of the docker compose file. I would advocate for changing the contribute guide and tell people to set up the environment with a docker compose directly. It is more straightforward from my point of view.
